src/deploy/osp_deployer/settings/gen_scripts/gen_prop_file.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/python
+
+# Copyright (c) 2018 Dell Inc. or its subsidiaries.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Run this script run from the director node as the director's admin user.
+# This script assumes the update_ssh_config.py is present.
+###############################################################################
+
+# IMPORTS
+import csv
+import json
+import sys
+import io
+import os
+
+from collections import defaultdict
+
+
+class parse_key_components:
+    role_type = None
+    node_id = None
+
+    def setKey(self, key):
+        key_tokens = key.split('_')
+        key_token_count = len(key_tokens)
+        if key_token_count > 2:
+            self.node_id = key_tokens[1]
+            self.role_type = key_tokens[0]
+        else:
+            self.role_type = key_tokens[0]
+
+    def getNodeId(self):
+        if unicode(self.node_id).isnumeric():
+            return self.node_id
+        else:
+            return 'false'
+
+    def getRoleType(self):
+        return self.role_type
+
+
+def read_input_file(input_file):
+    inputFile = open("exported_name_value.csv", 'r')
+    csvIn = csv.DictReader(inputFile, fieldnames=("Key", "Value"),
+                           delimiter='|')
+
+    settings = {}
+    keys = []
+    value = []
+
+    for row in csvIn:
+        if row["Key"] not in keys:
+            key = row["Key"]
+            value = row["Value"]
+            if row["Key"].startswith("sah_"):
```

Comment:
From this point on, why not just use 'key' in place of row["Key"]?  it's already been assigned to key.